src/CycloneDX.Core/Models/CryptoProperties.cs

```diff
@@ -0,0 +1,296 @@
+﻿using ProtoBuf;
+using System;
+using System.Collections.Generic;
+using System.Net.Sockets;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Xml.Serialization;
+
+namespace CycloneDX.Core.Models
+{
+    [XmlType("cryptoProperties")]
+    [ProtoContract]
+    public class CryptoProperties
+    {
+        [XmlElement("assetType")]
+        [ProtoMember(1)]
+        public AssetType AssetType { get; set; }
+        [XmlElement("algorithmProperties")]
+        [ProtoMember(2)]
+        public AlgorithmProperties AlgorithmProperties { get; set; }
+        [XmlElement("certificateProperties")]
+        [ProtoMember(3)]
+
+        public CertificateProperties CertificateProperties { get; set; }
+        [XmlElement("relatedCryptoMaterialProperties")]
+        [ProtoMember(4)]
+        public RelatedCryptoMaterialProperties RelatedCryptoMaterialProperties { get; set; }
+        [XmlElement("protocolProperties")]
+        [ProtoMember(5)]
+        public ProtocolProperties ProtocolProperties { get; set; }
+        [XmlElement("oid")]
+        [JsonPropertyName("oid")]
+        [ProtoMember(6)]
+        public string ObjectIdentifier { get; set; }
+    }
+
+    [ProtoContract]
+    public class ProtocolProperties
+    {
+        [XmlElement("type")]
+        [ProtoMember(1)]
+        public ProtocolType Type { get; set; }
+
+        [XmlElement("version")]
+        [ProtoMember(2)]
+        public string Version { get; set; }
+
+        [XmlArray("cipherSuites")]
+        [XmlArrayItem("cipherSuite")]
+        [ProtoMember(3)]
+        public List<CipherSuite> CipherSuites { get; set; }
+
+        [XmlElement("ikev2TransformTypes")]
+        [ProtoMember(4)]
+        public Ikev2TransformTypes Ikev2TransformTypes { get; set; }
+    }
+
+    public enum ProtocolType
+    {
+        Null,
+        [XmlEnum("unknown")]
+        Unknown,
+        [XmlEnum("other")]
+        Other,
+        [XmlEnum("tls")]
+        Tls,
+        [XmlEnum("ssh")]
+        Ssh,
+        [XmlEnum("ipsec")]
+        Ipsec,
+        [XmlEnum("ike")]
+        Ike,
+        [XmlEnum("sstp")]
+        Sstp,
+        [XmlEnum("wpa")]
+        Wpa
+    }
+
+    [ProtoContract]
+    public class Ikev2TransformTypes
```

Comment:
@mtsfoni I'd bet that the JSON serialization is wrong here.

Reply:
Should be fixed by https://github.com/CycloneDX/cyclonedx-dotnet-library/pull/355